### PR TITLE
Added description support to beteleguse

### PR DESCRIPTION
**Pull request by Akasurde**
Fixes : https://github.com/SatelliteQE/betelgeuse/issues/44

Signed-off-by: Abhijeet Kasurde akasurde@redhat.com

### CODE REVIEW

**Comment by Akasurde** [comment]
@elyezer Could you please check changes ? Thanks,

File: betelgeuse.py [hunk 1]
```diff
@@ -248,6 +248,8 @@ def add_test_case(args):
                 test.docstring = test.docstring.decode('utf8')
             test.docstring = RST_PARSER.parse(test.docstring)
 
+        test_desc = test.unexpected_tags.get('description', test.docstring if test.docstring else '')
```
Comment (elyezer) [inline betelgeuse.py]: Testimony have been update and now the information should be retrieved from the `test.tokens` instead of `test.unexpected_tags`. Also you need to rebase and make sure to add the description to the list of testimony.SETTINGS['tokens'].

**Comment by elyezer** [comment]
ACK, pending comment.

I have not looked before because the WIP on the title. When you send a PR that needs a pre-review please make sure to state it like you did when tagged my name. That will help identify if it is ready for the pre-review or something is about to come before that.

**Comment by elyezer** [comment]
The rebase is to incorporate c4676ce6d9062a8757ea1c83f89ae8845a698099

File: betelgeuse.py (hunk 1 above)
Comment (Akasurde) [inline betelgeuse.py]: OK.

**Comment by Akasurde** [comment]
@elyezer I will rebase changes and re-submit PR. I will keep in mind about recommendation. Thanks,

File: betelgeuse.py [hunk 2]
```diff
@@ -248,7 +248,12 @@ def add_test_case(args):
         if test.docstring:
             if not type(test.docstring) == unicode:
                 test.docstring = test.docstring.decode('utf8')
-            test.docstring = RST_PARSER.parse(test.docstring)
+            test.docstring = RST_PARSER.parse(str(test.docstring))
```
Comment (elyezer) [inline betelgeuse.py:251]: If you do this the decode operation above will be undone. Why is this needed?

File: betelgeuse.py [hunk 3]
```diff
@@ -248,7 +248,12 @@ def add_test_case(args):
...
+
+        test_desc = test.tokens.get(
```
Comment (elyezer) [inline betelgeuse.py:253]: Please name this variable as the Polarion field, `description`. In this case.

**Comment by elyezer** [comment]
@Akasurde since this PR have been open for a long time, I will handle the above comments and merge. Thank you

File: betelgeuse.py (hunk 2 above)
Comment (Akasurde) [inline betelgeuse.py:251]: I fixed this as `make test` failed with error 

```
TypeError: coercing to Unicode: need string or buffer
```

File: betelgeuse.py (hunk 3 above)
Comment (Akasurde) [inline betelgeuse.py:253]: OK

**Comment by elyezer** [comment]
Merged as 7a5ceec74680a65763e9fe286669d5dd528077c4.

**Comment by Akasurde** [comment]
@elyezer Thanks.
